src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/scope_map.py

```diff
@@ -0,0 +1,159 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+from azure.cli.core.util import CLIError
+from ._utils import get_resource_group_name_by_registry_name
+
+
+def _validate_and_generate_actions_for_repositories(allow_or_deny_respository):
```

Comment:
put this method in `_utils.py`, and add some comments and necessary `None` checks

Reply:
Even though this is only used by `scope_map.py`?

Reply:
What necessary `None` checks are you suggesting?

Reply:
Yeah , I guess this might just be useful in this file then. I will defer to @djyou

Reply:
using `_generate_actions_from_repositories`

Reply:
actually using `_parse_actions_from_repositories` which makes more sense :)